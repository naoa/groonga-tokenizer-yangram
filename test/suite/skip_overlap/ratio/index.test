register tokenizers/yangram

config_set tokenizer.yangram.Terms.skip_overlap_ratio 0.5

table_create Terms TABLE_HASH_KEY ShortText \
  --default_tokenizer TokenYaBigram \
  --normalizer NormalizerAuto

table_create Memos TABLE_NO_KEY
column_create Memos title COLUMN_SCALAR ShortText
column_create Memos content COLUMN_SCALAR ShortText

load --table Memos
[
{"title": "今日", "content": "今日は雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"}
]

column_create Terms memos_content COLUMN_INDEX|WITH_POSITION Memos content
column_create Terms @memos_title COLUMN_INDEX|WITH_POSITION Memos title

table_tokenize Terms "今日は雨だな" --mode GET
