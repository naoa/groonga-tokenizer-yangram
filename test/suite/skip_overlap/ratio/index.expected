register tokenizers/yangram
[[0,0.0,0.0],true]
config_set tokenizer.yangram.Terms.skip_overlap_ratio 0.5
[[0,0.0,0.0],true]
table_create Terms TABLE_HASH_KEY ShortText   --default_tokenizer TokenYaBigram   --normalizer NormalizerAuto
[[0,0.0,0.0],true]
table_create Memos TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Memos title COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
column_create Memos content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
load --table Memos
[
{"title": "今日", "content": "今日は雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"},
{"title": "今日", "content": "明日が雨だな"}
]
[[0,0.0,0.0],6]
column_create Terms memos_content COLUMN_INDEX|WITH_POSITION Memos content
[[0,0.0,0.0],true]
column_create Terms @memos_title COLUMN_INDEX|WITH_POSITION Memos title
[[0,0.0,0.0],true]
table_tokenize Terms "今日は雨だな" --mode GET
[
  [
    0,
    0.0,
    0.0
  ],
  [
    {
      "value": "今日",
      "position": 0,
      "force_prefix": false
    },
    {
      "value": "日は",
      "position": 1,
      "force_prefix": false
    },
    {
      "value": "は雨",
      "position": 2,
      "force_prefix": false
    },
    {
      "value": "雨だ",
      "position": 3,
      "force_prefix": false
    },
    {
      "value": "だな",
      "position": 4,
      "force_prefix": false
    }
  ]
]
